src/main/java/net/snowflake/client/jdbc/RestRequest.java

```diff
@@ -346,6 +349,8 @@ public static CloseableHttpResponse execute(
         }
 
         retryCount++;
+        lastStatusCodeForRetry =
+            response == null ? "0" : String.valueOf(response.getStatusLine().getStatusCode());
```

Comment:
Thanks for making the change.
A few questions:
1. Is it intended to add "retry" query parameter on all requests from JDBC? 
    I think the asks was to (a) make sure all connectors sets the existing "retry" query parameter correctly when hitting /queries/v1/query-request endpoint, and (b) add a "retryReason" query parameter when hitting the same endpoint. 
    It looks like this PR only modifies JDBC and it sets "retry" query parameter but not "retryReason" parameter, which I think is a bit different from the original ask.
2. According to source code of GS's [/queries/v1/query-request endpoint](https://sourcegraph.c2.us-central1.gcp-dev.app.snowflake.com/github.com/snowflakedb/snowflake/-/blob/GlobalServices/src/main/java/com/snowflake/resources/QueriesResource.java?L666:19), the endpoint expects the value of "retry" parameter to be "true" or "false". However, this PR sets the value of "retry" parameter to the last request's status code. Even if we update the logic on the GS side, I think we will have backward compatibility issue, unless we add a new  /queries/**v2**/query-request endpoint, which I don't think is necessary for our purpose.

Reply:
@sfc-gh-twang this has been updated to rename the query parameter to "retryReason" instead of just "retry"

Reply:
@sfc-gh-mknister 
Thanks for adding retryReason query parameter!
A few questions:
The PR only modifies JDBC, right? 
Does JDBC now set the retry  and retryCount query parameters?

I think the original asks were to 
(a) make sure all connectors sets the existing "retry" and "retryCount" query parameters correctly when hitting /queries/v1/query-request endpoint, and 
(b) add a "retryReason" query parameter when hitting the same endpoint.

I think your PR did (b), but not sure about (a).
And I was wondering what about other connectors, are they going to come in other PRs and this is just the first PR?

Reply:
@sfc-gh-twang to answer your questions:

This PR only modifies JDBC. Each connector is in a different repository so each driver will have a separate PR for this task.

The JDBC already sets "retryCount" when issuing a retry. It does not and has never set the boolean query parameter "retry". Looking at the GS code logic, I don't think that "retry" needs to be set to true as long as "retryCount" is incremented. The only place "retry" is used is in this line, and it's in an OR clause with "retryCount > 0". https://github.com/snowflakedb/snowflake/blob/main/GlobalServices/src/main/java/com/snowflake/resources/QueriesResource.java#L707 Is there a reason that both should be set?

This PR adds the query parameter retryReason, which will provide an integer value of the last HTTP status code returned from GS from the previous response. If the retry is due to a socket timeout and there is no status code available, the retryReason will be set to 0.